gate-web/src/main/groovy/com/netflix/spinnaker/gate/services/SessionService.groovy

```diff
@@ -0,0 +1,43 @@
+  /*
+   * Copyright 2015 Netflix, Inc.
+   *
+   * Licensed under the Apache License, Version 2.0 (the "License");
+   * you may not use this file except in compliance with the License.
+   * You may obtain a copy of the License at
+   *
+   *    http://www.apache.org/licenses/LICENSE-2.0
+   *
+   * Unless required by applicable law or agreed to in writing, software
+   * distributed under the License is distributed on an "AS IS" BASIS,
+   * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   * See the License for the specific language governing permissions and
+   * limitations under the License.
+   */
+
+  package com.netflix.spinnaker.gate.services
+
+  import groovy.transform.CompileStatic
+  import org.springframework.beans.factory.annotation.Autowired
+  import org.springframework.stereotype.Component
+  import redis.clients.jedis.Jedis
+  import redis.clients.jedis.JedisPool
+
+  @Component
+  @CompileStatic
+  class SessionService {
```

Comment:
Are you able to rewrite this in Java please?

Reply:
And the tests too if you don't mind.  We're really trying to avoid new groovy code wherever possible.

Reply:
Yep can do, just out of interest why is this? Is Java also preferred over Kotlin?

Reply:
Groovy allows too much loose/lazy typing.  Kotlin is fine.  It's more my personal preference for tests in java over kotlin because junit seems way more stable.  Like, IntelliJ no longer supports the version of [spek](https://github.com/spekframework/spek), and the project appears to be kinda dead.